File: CPP/Program/Motorcycle.cpp
//Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
//Memasukan library yang digunakan, termasuk kelas induk (kelas Vehicle) dari kelas Motorcycle
#include <iostream>
#include <string>
#include "Vehicle.cpp"


//Deklarasi namespace
using namespace std;


/*
    Kelas Motorcycle yang digunakan untuk merepresentasikan data Motorcycle,
    kelas Motorcycle merupakan kelas turunan dari kelas induk Vehicle.
*/
class Motorcycle : public Vehicle
{
private:
    //Atribut yang digunakan pada kelas Motorcycle terdiri dari jenis motor dan kapasitas tangki
    string jenisMotor;
    string kapasitasTangki;
public:
    //constructor kosong
    Motorcycle()
    {


    }

    //contructor yang diisi parameter
    Motorcycle(string jenisMotor, string kapasitasTangki, string platNomor, string merk, string tahunProduksi, string warna)
    : Vehicle(platNomor, merk, tahunProduksi, warna)
    {
    this->jenisMotor = jenisMotor;
    this->kapasitasTangki = kapasitasTangki;
    }


    //membuat getter setter untuk jenis motor dan kapasitas tangki
    void setjenisMotor(string jenisMotor)
    {
        this->jenisMotor = jenisMotor;
    }


    void setKapasitasTangki(string kapasitasTangki)
    {
        kapasitasTangki = kapasitasTangki;
    }


    //Method get untuk mendapatkan nilai setiap atribut pada kelas Motorcycle
    string getJenisMotor()
    {
        return jenisMotor;
    }


    string getKapasitasTangki()
    {
        return kapasitasTangki;
    }

    //membuat fungsi untuk menampilkan jenis motor dan kapasitas tangki
    void display() const override {
        Vehicle::display();
        cout << "Jenis Motor: " << jenisMotor << endl;
        cout << "Kapasitas Tangki: " << kapasitasTangki << endl;
    }
    

    //Destructor
    ~Motorcycle()
    {


    }
};

File: Python/Program/Motorcycle.py
#Vehicle import Vehicle
from Vehicle import Vehicle
class Motorcycle(Vehicle):
    def __init__(self, jenis_motor="", kapasitas_tangki="", plat_nomor="", merk="", tahun_produksi="", warna=""):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)  #memanggil konstruktor kelas induk
        self.jenis_motor = jenis_motor  #inisialisasi atribut jenis_motor
        self.kapasitas_tangki = kapasitas_tangki  #inisialisasi atribut kapasitas_tangki

    def set_jenis_motor(self, jenis_motor):  #metode untuk mengatur jenis_motor
        self.jenis_motor = jenis_motor

    def set_kapasitas_tangki(self, kapasitas_tangki):  #metode untuk mengatur kapasitas_tangki
        self.kapasitas_tangki = kapasitas_tangki

    def get_jenis_motor(self):  #metode untuk mendapatkan jenis_motor
        return self.jenis_motor

    def get_kapasitas_tangki(self):  #metode untuk mendapatkan kapasitas_tangki
        return self.kapasitas_tangki

    def display(self):  #metode untuk menampilkan informasi kendaraan
        super().display()  #memanggil metode display dari kelas induk
        print("Jenis Motor:", self.jenis_motor)  #menampilkan jenis_motor
        print("Kapasitas Tangki:", self.kapasitas_tangki)  #menampilkan kapasitas_tangki
